src/sentry/api/endpoints/user_notification_providers.py

```diff
@@ -0,0 +1,84 @@
+from django.db import router, transaction
+from rest_framework import status
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.base import control_silo_endpoint
+from sentry.api.bases.user import UserEndpoint
+from sentry.api.exceptions import ParameterValidationError
+from sentry.api.serializers import serialize
+from sentry.api.validators.notifications import validate_type
+from sentry.models import User
+from sentry.models.notificationsettingprovider import NotificationSettingProvider
+from sentry.notifications.serializers import NotificationSettingsProviderSerializer
+from sentry.notifications.types import NotificationSettingsOptionEnum
+from sentry.notifications.validators import (
+    UserNotificationSettingsProvidersDetailsSerializer,
+    allowed_providers,
+)
+
+
+@control_silo_endpoint
+class UserNotificationProvidersEndpoint(UserEndpoint):
```

Comment:
i'm kind of confused on naming. this endpoint is `UserNotificationProvidersEndpoint`. but it serializes `NotificationSettingProvider`. from the endpoint name, i'd assume it returns a list of notification providers for the user, but, it actually returns settings for each provider?

Reply:
@JoshFerge should I put `settings` in the endpoint name? I was just trying to make the names a bit more concise.

Reply:
from an outsiders view it reduces complexity a tad. there are a lot of names and concepts and trying to get familiar with them all is hard -- lining up the endpoint names to the serialized object names helps a tad.